rsocket-core/src/main/java/io/rsocket/core/RSocketConnector.java

```diff
@@ -395,6 +397,23 @@ public RSocketConnector lease(Supplier<Leases<? extends LeaseStats>> supplier) {
     return this;
   }
 
+  /**
+   * When this is set, frames reassembler control maximum payload size which can be reassembled.
+   *
+   * <p>By default this is not set in which case maximum reassembled payloads size is not
+   * controlled.
+   *
+   * @param maxReassemblySize the threshold size for reassembly, must be no less than 16,777,215
+   * @return the same instance for method chaining
+   * @see <a
+   *     href="https://github.com/rsocket/rsocket/blob/master/Protocol.md#fragmentation-and-reassembly">Fragmentation
+   *     and Reassembly</a>
+   */
+  public RSocketConnector reassemble(int maxReassemblySize) {
+    this.maxReassemblySize = ReassemblyDuplexConnection.assertMaxReassemblySize(maxReassemblySize);
+    return this;
+  }
+
```

Comment:
I think this should apply to any input payload size regardless of whether it is fragmented or not. In fact by default fragmentation is off and in any case once there is a limit in place for fragmented input, it would be expected to have the same for non-fragmented input. I do understand that the rest is at the transport level. Nevertheless from an application perspective, conceptually, it's one limit.

Reply:
That is another concern. Spec specifies that Max-Frame size if 16MB, thus it should be clear that RSocket should support that size. Now the question is - if one applies a 1byte memory limit, how a client, which expects that allowed frame size is 16MB, should behave in that case.

From the fragmentation perspective, minimal frame size is 64bytes, so reassembly limit should be more than that at least, but if we wanna send something bigger than that, all tries will fail, so what is the benefit of such reassembly then? 

From my point of view, the minimal storage per stream should be at least the default max frame size. Or at least minimal fragment size specified by spec